src/librustdoc/html/static/js/search.js

```diff
@@ -154,13 +389,101 @@ window.initSearch = function(rawSearchIndex) {
             return NO_TYPE_FILTER;
         }
 
-        var valLower = query.query.toLowerCase(),
-            val = valLower,
-            typeFilter = itemTypeFromName(query.type),
-            results = {}, results_in_args = {}, results_returned = {},
-            split = valLower.split("::");
+        userQuery = userQuery.trim();
+        var query = {
+            original: userQuery,
+            userQuery: userQuery.toLowerCase(),
+            length: userQuery.length,
+            pos: 0,
+            typeFilter: null,
+            elems: [],
+            elemName: null,
```

Comment:
What is the purpose of `elemName`? Looking at the code it appears to also be internal state.